wordpress/web/app/plugins/owid/owid.php

```diff
@@ -307,7 +306,7 @@ function build_static($post_ID, $post_after, $post_before)
         $cmd =
             "cd " .
             ABSPATH .
-            "codelink && yarn runPostUpdateHook " .
```

Comment:
A little hacky, but I see it solves the problem!

Just to double-check, is this something that only ever runs on the server?

Reply:
Alternatives would be:

- Trying to put NVM settings into `~/.bash_profile` to see if the non-interactive shell picks them up there
- Picking the specific node version and actually specifying it by full path `/home/owid/.nvm/versions/node/v18.16.0/bin/yarn`
- Symlinking `/usr/local/bin/yarn` to the one we want

I have a modest preference for symlinking over sourcing the `.bashrc`, but it's only modest.

Reply:
> Just to double-check, is this something that only ever runs on the server?

As opposed to CI or local envs? It doesn't run on CI since this is a Wordpress hook. Might run on local envs with some degree of success if devs went through the trouble of - among other things - setting up `codelink` (which is unlikely).

Reply:
> Symlinking /usr/local/bin/yarn to the one we want

@larsyencken Isn't that going to break whenever we upgrade the node version?

Reply:
> Trying to put NVM settings into ~/.bash_profile to see if the non-interactive shell picks them up there

Is there an advantage over the early return for non-interactive shells in .bashrc?

Reply:
Oh, it's mainly a question of whether .bash_profile would get automatically sourced in this scenario or not. I'm not 100% sure, ChatGPT is not sure either.